internal/pkg/v2/infrastructure/redis/intervalaction.go

```diff
@@ -68,3 +68,26 @@ func addIntervalAction(conn redis.Conn, action models.IntervalAction) (models.In
 
 	return action, edgeXerr
 }
+
+// allIntervalActions queries intervalActions by offset and limit
+func allIntervalActions(conn redis.Conn, offset, limit int) (intervalActions []models.IntervalAction, edgeXerr errors.EdgeX) {
+	end := offset + limit - 1
+	if limit == -1 { //-1 limit means that clients want to retrieve all remaining records after offset from DB, so specifying -1 for end
+		end = limit
+	}
+	objects, edgeXerr := getObjectsByRevRange(conn, IntervalActionCollection, offset, end)
+	if edgeXerr != nil {
+		return intervalActions, errors.NewCommonEdgeXWrapper(edgeXerr)
+	}
+
+	intervalActions = make([]models.IntervalAction, len(objects))
+	for i, o := range objects {
+		dto := dtos.IntervalAction{}
+		err := json.Unmarshal(o, &dto)
```

Comment:
I assume this is where you need the latest core contract update to avoid using DTO?

Reply:
Oh, I forgot it, thank you for catch that! 
Modified and upgrade core-contracts lib.